osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -317,8 +317,8 @@ private void updateScore()
 
         protected virtual double ComputeTotalScore(double comboProgress, double accuracyProgress, double bonusPortion)
         {
-            return 700000 * comboProgress +
-                   300000 * Math.Pow(Accuracy.Value, 10) * accuracyProgress +
+            return 700000 * Accuracy.Value * comboProgress +
```

Comment:
Having accuracy multiplied in here is going to take me a while to get my head around, and accept on a conceptual level.

What happens once we show the individual score portions on the results screen? How do we tell a user that combo score can be less than max even with a full combo?

This may call for making this algorithm slightly more complex in order to keep these two pieces isolated..

Reply:
you can factor out one `Accuracy.Value` from both terms like this

```cs
return Accuracy.Value *
        (700000 * comboProgress + 300000 * Math.Pow(Accuracy.Value, 7) * accuracyProgress)
        + bonusPortion;
```

now you have combo and accuracy portions isolated, but you just multiply by the accuracy at the end (before bonus), which can be conveyed quite easily i think

Reply:
That doesn't help when displaying to a user.

To recap, I definitely want to be able to show at the results screen something like this:

![Safari 2023-08-22 at 09 52 17](https://github.com/ppy/osu/assets/191335/c9480ad4-f16a-4a41-9d25-e370ace59d25)

So the user can better understand the score breakdown.

Reply:
I don't know what's the best. I don't think factoring out one `Accuracy.Value` as @Walavouchey makes it any simpler.

First note that, just like in score v2 (and v1), `comboProgress` is 1 only if you SS (FC is not enough), because it accounts for the value of each judgements thus already factoring some accuracy (which is good).
Multiplying the final `comboProgress` by the acc was done to give more "importance" to accuracy on low acc scores (which is currently an issue in score v2) and thus bringing down a 70% acc FC to ~360k compared to the ~500k in v2.

I see 4 options:
- Keep the formula like that, and work on a way to better explain things. (Rename ComboScore in MixedScore ? idk)
- Keep the end-results the same, but have ComboScore split into a "ComboScore" without accuracy multiplier and some "AccComboMalus" whose value is simply the difference (`AccComboMalus = 700000 * comboProgress * (acc - 1)` which is negative).
- Change the formula to split score into 3 parts instead of 2 : ComboScore (no acc multiplier), AccScore, and MixedScore, where ComboScore has no acc multiplier but MixedScore would something like `comboProgress * Pow(Accuracy.Value, 3)`. The best proportions (300k/300k/400k ? 333k each ?) of the three and acc exponents would have to be studied.
- Simply revert that part of the change, and go back to v2's acc scaling (The result being "Square-root combo with original acc scaling of score v2" in the spreadsheet). This is the simplest but then it wouldn't address the above mentioned issue with low-acc scores. Since this low-acc scores issue could be considered an "edge case", I would personally find it slightly worse but still acceptable though.

Which one do you prefer ?

Reply:
> I don't know what's the best. I don't think factoring out one Accuracy.Value as @Walavouchey makes it any simpler indeed.

Agree

> Which one do you prefer ?

The only valid one if we want to aim for the acc/combo split display on the results screen is the last option, IMO. That said, if I'm the only one that wants to see this information on the results screen and the majority opinion is that it's not useful, I'm willing to go with option 1 and do away with that concept altogether.

@ppy/team-client interested in other opinions here. and players that may happen to be watching this issue.

Reply:
I think the solution proposed by @Walavouchey isn't this unreasonable.
By factoring out the accuracy part from the combo portion you could display the score like this :

![SqrtScore Ranking Panel](https://github.com/ppy/osu/assets/52672543/a48fd31a-506a-4df4-8075-c1194239baaf)

Sure, it's 3 values instead of 2 and isn't as easy to understand, but I could see a simple tooltip on hover of "accuracy multiplier" work nicely to give further explanation if needed.

Reply:
I thought breaking down the score at the results screen was a cool idea and was looking forward to seeing it. It would also help with cases like [the one mentioned in Discord](https://discord.com/channels/188630481301012481/1130086080687919115/1130185355946098829). 
Can't comment on the importance of fixing the "low acc scores issue". I didn't know that was an issue.

Reply:
I think the second option provided by @Zyfarok is the best to provide simpler explanation while maintaining the score change. Since it depends on the subtraction of a variable, it seems easier to understand at first sight and to represent on the results screen. We could name it something like "inaccuracy penalty" and put a tooltip explaining the origin of it. I understand that some players may not like it because it would seem like the total score depends on two variables based on accuracy against one based on combo.
Also, I don't understand much about score and programming. I'm just a lazer-only player following this pr, so I'm sorry if I misunderstood something.

Reply:
@Zyfarok can you comment on how important you see the accuracy fix being, maybe on a 1 to 10 scale? I tend to agree with @bdach that it's an edge case we probably don't need to worry about, and just overcomplicates things.

Reply:
I could say 5/10 but I don't really now how to place this on a scale, and what that scale would mean... @WitherFlower is telling me 7/10 and he wouldn't see that as an edge-case at all. It would probably be better to ask more community opinion (bubbleman?) but it's also really hard to get feedback of that level.

#### For changing the 70/30 mix as @bdach suggests:

We explored changing it to 60/40 or 50/50 while reducing the acc exponent a bit. This indeed increases the weight of acc on lower-acc scores while maintaining similar weight of acc on high-acc scores. The only thing which made us not fully convinced by it, is that it would further derive from the "general curve" of v2, in the sense that what people would usually judge as a "bad score" could give something like 200k in v2 while it would now give something above 400k if we go with a 50/50 split and the proper acc exponent. I believe this would however bring osu! closer to how scoring works in taiko/mania/ctb, so it is not necessarily a bad thing! It is simply that because it's further away from what people from osu-the-ruleset are used to, it makes it harder to evaluate and thus feel less like a "safe" direction.

However, since the goal of merging this is also to gather people's opinion on that scoring, it might be good to go with something as "extreme" as the 50/50 mix and see what people think of it first, and go back to the 70/30 mix proposed here in case it is seen as problematic.

I've added a "alternative 4" version in here with a 50/50 mix : https://docs.google.com/spreadsheets/d/1g-hel7ywUdZFQGpaC5UVeURBYLuPhTS6rTnI1-ImoTI/edit#gid=0
This would give:
```cs
return (long)Math.Round((
    500000 * comboProportion
    + 500000 * Math.Pow(score.Accuracy, 5)
    + bonusProportion) * modMultiplier);
```

Do you want to try with that instead ?

Note that it still "accounts for accuracy" in the `comboProportion` because of `GetComboScoreChange` using `Judgement.ToNumericResult(result.Type)` as in v1/v2, thus an FC still wouldn't give 500k ComboScore. I think that it is a good think but is this something you would also want changed ? @peppy

Reply:
A problem I see with changing the formula away from v2 any further is that it'll likely break the balance with classic scoring i showed in https://github.com/ppy/osu/issues/23763#issuecomment-1689533131, meaning that some sacrifices will have to be made on that front. 

To give my final opinion as to all of the options that were proposed, I don't think not awarding 700,000 score for an FC is that big of a deal (SV2 already does that btw, as the hit value is multiplied by the combo on each hit meaning you have to SS to actually get maximum combo score). 

It seems to me that we're just trading game balance for a little bit more "simplicity" for a system that is already hard to understand anyway, so I'm not sure it helps much. Given how much scoring matters in competitive play, I don't think that tradeoff is worth it.

Reply:
Let me say one thing then: I'm not willing to do away with having definite score components. So the existing state of the PR *cannot work*.

A good starting point would be to switch back to `Square-root combo with original acc scaling of score v2` IMO.

We could also remove the "alternatives" which change this from the sheet to simplify what people are looking at.

I'm not against changing the split of combo/acc, but I'm also still not sure, looking at the graphs, whether it is even required.

Reply:
> I'm not willing to do away with having definite score components.

Does that mean that a **requirement** of scoring is to give maximum combo score in case of a full combo, no matter the accuracy ?

Reply:
correct

Reply:
Ok, so if you want an FC to give the full ComboScore no matter the acc, (I'm repeating myself just in case):
**That's not how score v2 works** (and thus not what `Square-root combo with original acc scaling of score v2` gives neither).

I believe it's not the correct direction, but we could try, so here's what we could do:

First, this means `GetComboScoreChange` could simply become:
```cs
protected virtual double GetComboScoreChange(JudgementResult result) => Judgement.ToNumericResult(result.Judgement.MaxResult) * Math.Pow(result.ComboAfterJudgement, 0.5);
```

Then, it can not stay 70/30 and definitely needs to lean towards AT LEAST 500k towards AccScore. Otherwise there's no way to give a reasonable weight to accuracy in the very reasonable "low acc" scenarios (~80%). Thus, a first sketch of reasonable example solution could be:
```
return (long)Math.Round((
    500000 * comboProportion
    + 500000 * Math.Pow(score.Accuracy, 7)
    + bonusProportion) * modMultiplier);
```

Same question as above: Do you want to try with that instead ?

Again, this is very far from the original progression curve of v2. It might be fine (Except for score-farming as Wither mentioned, but it's quite a niche application of scoring) but it would be nice to test that in team versus tournaments where the effect of that is the most prominent. (Example: A 2v2 scenario with 1x good player + 1x bad player versus 2x intermediate players)

Reply:
Also about:
> No one has complained until now about this scenario.

v2 has almost only be used in tournament (and tournament training) until now.
Low accs FCs at the scale of 70% rarely happen in tournament in practice, and when they happen, the match is rarely any close due to v2's ComboScore scaling. I believe this issue has thus been overshadowed. This issue is the conclusion I ended up when I asked a small sample (about 10 people) of the tournament community how much score a low acc FCs should give as well as how it should compare to a 1 miss with higher accuracy.

If you want to gather more opinions on the question, then this is the kind of poll that we should send to the community of experienced players. I can prepare a list of question for such a poll if you think that would help.

Reply:
I think a starting point would be showing which leaderboards have accuracy below 70% in the top 500

Reply:
https://twitter.com/ppy/status/1696478639679037934

Reply:
Not polling on twitter because no one can agree on anything and the resulst are always wrong.